commit b4844472ec43de3786401c718923f2aac8f3a5d4 (refs/changes/26/526/3)
Author: Trent Mick <trentm@gmail.com>
Date:   2016-09-23T22:05:14+00:00 (3 years ago)
    
    TOOLS-1550 'make cutarelease' for node-sdc-clients to encourage tag and publishing on new revs
    Reviewed by: Todd Whiteman <todd.whiteman@joyent.com>

diff --git a/Makefile b/Makefile
index 5b8f1f9..3040e4a 100644
--- a/Makefile
+++ b/Makefile
@@ -83,5 +83,31 @@ test: ca_test ufds_test cnapi_test napi_test vmapi_test papi_test cns_test
 setup:
 	$(NPM) install
 
+# Ensure CHANGES.md and package.json have the same version.
+.PHONY: versioncheck
+versioncheck:
+	@echo version is: $(shell cat package.json | json version)
+	[[ `cat package.json | json version` == `grep '^## ' CHANGES.md | head -1 | awk '{print $$2}'` ]]
+
+.PHONY: cutarelease
+cutarelease: versioncheck
+	[[ -z `git status --short` ]]  # If this fails, the working dir is dirty.
+	@which json 2>/dev/null 1>/dev/null && \
+	    ver=$(shell json -f package.json version) && \
+	    name=$(shell json -f package.json name) && \
+	    publishedVer=$(shell npm view -j $(shell json -f package.json name)@$(shell json -f package.json version) version 2>/dev/null) && \
+	    if [[ -n "$$publishedVer" ]]; then \
+		echo "error: $$name@$$ver is already published to npm"; \
+		exit 1; \
+	    fi && \
+	    echo "** Are you sure you want to tag and publish $$name@$$ver to npm?" && \
+	    echo "** Enter to continue, Ctrl+C to abort." && \
+	    read
+	ver=$(shell cat package.json | json version) && \
+	    date=$(shell date -u "+%Y-%m-%d") && \
+	    git tag -a "v$$ver" -m "version $$ver ($$date)" && \
+	    git push --tags origin && \
+	    npm publish
+
 include ./tools/mk/Makefile.deps
 include ./tools/mk/Makefile.targ
